packages/webex/src/calling.js

```diff
@@ -68,6 +70,67 @@ class Calling extends EventEmitter {
       });
   }
 
+  async deregister() {
+    if (!this.registered) {
+      this.log.info('Authentication: webex.internal.device.deregister already done', logContext);
+
+      return Promise.resolve();
+    }
+
+    const promises = [];
+
+    const lineIds = Object.keys(this.callingClient?.getLines());
+
+    const lines = lineIds.length ? this.callingClient.getLines() : {};
+    for (const lineId of lineIds) {
```

Comment:
Why can't we run the loop directly on the lines array? Why do we need to maintain another array of lineIds?

Reply:
- lines isn't an array. It is an object that contains `{ lineId: lineObject }`
- When we do a for..in loop, it would go through prototype keys as well. Linter fails
- That is why we had to get lineIds in an array and then use it.

Reply:
getLines() API returns Record<string, ILine> so similar to how we are doing Object.keys() to fetch the keys which is lineId, we can do Object.values() which will return line array and then we can run for loop in the line array returned.